src/compiler/rust/canister_methods/src/upload_assets/mod.rs

```diff
@@ -0,0 +1,213 @@
+pub mod temp_chunks;
+
+use quote::quote;
+
+pub fn get_upload_assets() -> proc_macro2::TokenStream {
+    let temp_chunks_utils = temp_chunks::get_temp_chunk_utils();
```

Comment:
Do we like this approach? If so should we make it like this for the hashing as well?